modules/social_features/social_content_block/src/ContentBuilder.php

```diff
@@ -162,6 +172,13 @@ static function ($field_name) use ($block_content) {
             ->loadMultiple($entities);
 
           foreach ($entities as $key => $entity) {
+            // @todo Better to move to entity query check but
```

Comment:
@nechai could you explain more how did you try to make it better and why it didn't work?

Reply:
Agreed, this needs a bit more explaining on why this change is needed, also release notes are a bit vague. Ideally we would write tests for this, so we don't break it in the  future again.

Reply:
What I mean here that group visibility doesn't implemented on query level... To achieve this we should rewrite default group [query access handler](https://git.drupalcode.org/project/group/-/blob/8.x-1.x/src/Entity/Group.php#L45). This work require deeper investigation and refinement as can have critical impact on the whole distro.